torch/distributed/_tensor/sharding_prop.py

```diff
@@ -107,22 +109,45 @@ def _propagate_tensor_meta(self, op_schema: OpSchema) -> object:
             return None
 
     def _wrap_output_spec_tensor_meta(
-        self, output_spec: OutputSpecType, output_tensor_meta: object
+        self,
+        op: OpOverload,
+        output_spec: OutputSpecType,
+        output_tensor_meta: Union[TensorMeta, List[TensorMeta], Tuple[TensorMeta, ...]],
     ) -> None:
         """
         Wrap the output_spec with the tensor metadata from the output.
         """
+
         if output_spec is not None:
-            if isinstance(output_spec, DTensorSpec):
-                assert isinstance(output_tensor_meta, TensorMeta)
-                output_spec.tensor_meta = output_tensor_meta
-            elif isinstance(output_spec, (tuple, list)):
-                for i, spec in enumerate(output_spec):
-                    if isinstance(spec, DTensorSpec):
-                        assert isinstance(output_tensor_meta, (tuple, list))
-                        output_tensor_meta_i = output_tensor_meta[i]
-                        assert isinstance(output_tensor_meta_i, TensorMeta)
-                        spec.tensor_meta = output_tensor_meta_i
+            # Normalize
```

Comment:
Yep good point - I will change it. Another option would be to validate when the `OutputSharding` object is initialised. Do you think this would be possible? I couldn't quite work it out

Reply:
Maybe we can pass tensor_meta to OutputSharding init logic and attach the tensor meta there, let's do it in a separate PR